File: daily/Binary Search/275-H-index-ii.go
package daily

func hIndex(citations []int) int {
	n := len(citations)
	low, high := 0, n-1
	for low <= high {
		mid := (low + high) / 2
		h := n - mid
		if citations[mid] >= h {
			high = mid - 1
		} else {
			low = mid + 1
		}
	}
	return n - low
}

File: daily/Binary Search/275-H-index-ii.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        n = len(citations)
        low, high = 0, n - 1
        while low <= high:
            mid = (low + high) // 2
            h = n - mid
            if citations[mid] >= h:
                high = mid - 1
            else:
                low = mid + 1
        return n - low